app/API/V1/Entities/LoginAttempt.php

```diff
@@ -0,0 +1,325 @@
+
+    /**
+     * Returns updatedFromIp.
+     *
+     * @return string
+     */
+    public function getUpdatedFromIp()
+    {
+        return $this->updatedFromIp;
+    }
+
+    /**
+     * On an entity with HasLifecycleCallbacks it will run the special features of tt entities before persist
+     *
+     * @ORM\PreUpdate
+     * @throws \RuntimeException
+     */
+    public function ttPrePersist():void
+    {
+        //$arrayHelper = $this->getConfigArrayHelper();
+        //if ($arrayHelper !== NULL) {
+        /** @noinspection PhpParamsInspection */
+        //$arrayHelper->ttPrePersist($this);
+        //}
+    }
+
+    /**
+     * @return array
+     */
+    public function getTTConfig(): array
+    {
+        return [
+            'default'=>[
+                'create'=>[
+                    'allowed'=>false,
+                    'toArray'=> [
+                        'id'=>[],
+                        'count'=>[],
+                        'user'=>[],
+                    ]
+                ],
+                'update'=>[
+                    'extends'=>[':default:create'],
+                ],
+                'delete'=>[
+                    'extends'=>[':default:create'],
+                ],
+                'read'=>[ // Same as default create
+                    'extends'=>[':default:create']
+                ],
+            ],
+            'guest'=>[ // can do everything in default, and is allowed to do it when a super admin
+                'create'=>[
+                    'extends'=>[':default:create'],
+                    'allowed'=>true
+                ],
+                'update'=>[
+                    'extends'=>[':default:create'],
+                    'allowed'=>true
+                ],
+                'delete'=>[
+                    'extends'=>[':default:create'],
+                    'allowed'=>true
+                ],
+                'read'=>[ // Same as default create
+                    'extends'=>[':default:create']
+                ],
+            ],
+            'superAdmin'=>[ // can do everything in default, and is allowed to do it when a super admin
+                'create'=>[
+                    'extends'=>[':default:create'],
+                    'allowed'=>true,
+                    'notifications'=>[ // A list of arbitrary key names with the actual notifications that will be sent
+                        'fullLockCount'=>[
+                            'notification'=>new LoginLockNotification($this),
+                            'via'=>[
+                                'mail'=>[
+                                    'to'=>ArrayExpressionBuilder::closure(function () {
+                                        return $this->getUser()->getEmail();
+                                    })
+                                ]
+                            ]
+                        ]
+                    ]
+                ],
+                'update'=>[
+                    'extends'=>[':default:create'],
```

Comment:
Looks like instead of including the same settings as: superAdmin:create you could just extend superAdmin:create. That would remove any redundant code

Reply:
@jerazo

Reply:
Fixed